querygen/querygen.go

```diff
@@ -87,3 +87,25 @@ func GenClairScanCompletedQuery(index string) string {
 	)
 	return q
 }
+
+// GenReleaseCompletedQuery returns a Splunk query for generating Segment events
+// representing the Release resource success/failure state changes.
+func GenReleaseCompletedQuery(index string) string {
+	q, _ := UserJourneyQueryGen(
+		index,
+		`verb=patch `+
+			`"responseStatus.code"=200 `+
+			`"objectRef.apiGroup"="appstudio.redhat.com" `+
+			`"objectRef.resource"="releases" `+
+			`"objectRef.subresource"="status" `+
+			`"responseObject.metadata.resourceVersion"="*" `+
+			`"responseObject.status.completionTime"="*" `+
+			`| spath responseObject.status.conditions{} `+
+			`| mvexpand responseObject.status.conditions{} `+
+			`| search responseObject.status.conditions{}.type="Released" AND `+
+			`(responseObject.status.conditions{}.reason="Succeeded" OR responseObject.status.conditions{}.reason="Failed") `+
+			`| eval event="Release completed events"`,
+		[]string{"namespace", "application", "component", "release_status_message", "release_status_reason", "release_status_type"},
```

Comment:
`application` and `component` are always null. It's probably because the `Release` CRs don't have the respective annotations. Please look for any useful annotations in their place or omit them here.